### ISSUE TITLE
libPolicy.so is linked to executable when TELEMETRY_MONITOR is ON

### ISSUE THREAD

**Issue by shoamano83** [smartdevicelink/sdl_core#2320]
### Bug Report
The executable `smartDeviceLinkCore` has dependency on `libPolicy.so` when `TELEMETRY_MONITOR` configuration is enabled.

The executable loads `libPolicy.so` at runtime (please refer to `PolicyHandler::LoadPolicyLibrary()`). So IMHO, the library is not meant to be linked at build time.

##### Reproduction Steps
1. Check out latest `develop` branch (commit ba1da7bc2ff3fd37638620f6bea526384646ab0e)
2. Run `cmake`, `make` and `make install`
3. Under `bin/` directory, inspect dependency of the executable using `ldd`

##### Expected Behavior
I am not 100% sure but I think the executable should not have dependency on `libPolicy.so`.

##### Observed Behavior
The executable is linked to `libPolicy.so`:
```
$ ldd smartDeviceLinkCore 
	linux-vdso.so.1 =>  (0x00007fff845dd000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f9488d59000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f9488b55000)
	libbluetooth.so.3 => /usr/lib/x86_64-linux-gnu/libbluetooth.so.3 (0x00007f9488933000)
	libusb-1.0.so.0 => /lib/x86_64-linux-gnu/libusb-1.0.so.0 (0x00007f948871b000)
	liblog4cxx.so.10 => /usr/local/lib/liblog4cxx.so.10 (0x00007f9488335000)
	libapr-1.so.0 => /usr/local/lib/libapr-1.so.0 (0x00007f9488102000)
	libbson.so.0 => /usr/local/lib/libbson.so.0 (0x00007f9487efb000)
	libboost_system.so.1.66.0 => /usr/local/lib/libboost_system.so.1.66.0 (0x00007f9487cf7000)
	libPolicy.so => /home/samano/projects/sdl_core/cmake_build/bin/libPolicy.so (0x00007f9487985000)
	libcrypto.so.1.0.0 => /lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007f9487541000)
	libssl.so.1.0.0 => /lib/x86_64-linux-gnu/libssl.so.1.0.0 (0x00007f94872d8000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f94870d0000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f9486d4e000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f9486b38000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f948676e000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f9488f76000)
	libudev.so.1 => /lib/x86_64-linux-gnu/libudev.so.1 (0x00007f948914f000)
	libaprutil-1.so.0 => /usr/local/lib/libaprutil-1.so.0 (0x00007f9486548000)
	libemhashmap.so.0 => /usr/local/lib/libemhashmap.so.0 (0x00007f9486345000)
	libsqlite3.so.0 => /usr/lib/x86_64-linux-gnu/libsqlite3.so.0 (0x00007f9486070000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9485d67000)
	libexpat.so => /usr/local/lib/libexpat.so (0x00007f9485b33000)
	libcrypt.so.1 => /lib/x86_64-linux-gnu/libcrypt.so.1 (0x00007f94858fb000)
```

##### OS & Version Information
* OS/Version: Ubuntu 16.04 (amd64)
* SDL Core Version: Commit ba1da7bc2ff3fd37638620f6bea526384646ab0e
* Testing Against: N/A

##### Test Case, Sample Code, and / or Example App

Note that if I ran cmake with `-DTELEMETRY_MONITOR=OFF`, the dependency would be removed.

**Comment by shoamano83** [comment]
I think the cause is that `src/components/telemetry_monitor/CMakeLists.txt` specifies "Policy" in the LIBRARIES list:
https://github.com/smartdevicelink/sdl_core/blob/develop/src/components/telemetry_monitor/CMakeLists.txt#L65

I also checked with `4.4.1` tag and found that the executable had dependency on `libPolicy.so`. So I think this behavior has been around for a while.

**Comment by iCollin** [comment]
Closed as not reproducible against the first 7.1 release candidate. I no longer see libPolicy.so linked into smartDeviceLinkCore executable.

Please let me know if you believe this issue is still actual.
